sdk/identity/identity/src/credentials/interactiveBrowserCredential.browser.ts

```diff
@@ -108,24 +210,22 @@ export class InteractiveBrowserCredential implements TokenCredential {
       }
     }
 
-    let authPromise: Promise<msal.AuthResponse> | undefined;
+    let authPromise: MSALAuthenticationPromise | undefined;
+
     if (authResponse === undefined) {
       logger.info(
         `Silent authentication failed, falling back to interactive method ${this.loginStyle}`
       );
       switch (this.loginStyle) {
         case "redirect":
-          authPromise = new Promise((resolve, reject) => {
-            this.msalObject.handleRedirectCallback(resolve, reject);
-          });
+          authPromise = this.msalObject.handleRedirectPromise();
           this.msalObject.acquireTokenRedirect(authParams);
+          authResponse = (await authPromise) || undefined;
```

Comment:
`acquireTokenRedirect` will navigate away from the page, so it is recommended to not have any code running after this invocation to prevent unintended side effects.